packages/cojson/src/sync.ts

```diff
@@ -511,36 +464,34 @@ export class SyncManager {
       }
     }
 
-    if (entry.state.type === "unknown") {
+    // When we receive a known state from a peer and the header is "false"
+    // it means that the peer doesn't know anything about this coValue
+    // So we can say that the coValue has been not found on this peer
+    if (entry.state.type !== "available") {
       const availableOnPeer = peer.optimisticKnownStates.get(msg.id)?.header;
 
```

Comment:
This line is familiar to me as the failing invitation flow consists of two steps of handling two messages in a row - the known one without a header and the content one with a header. I thought the lacking header in the known was a "feature" but now I see that actually the header is expected for the value well known on a peer. 
Not sure if this PR solves the lack of the header in the known msg. I probably need to look into that one more time

Reply:
> I thought the lacking header in the known was a "feature" but now I see that actually the header is expected for the value well known on a peer.

The header is a boolean value that tells us if the other peer do have information about the header.
If it's false in this point it means that the coValue is unavailable on the other peer.

In the case of the invitation flow, if we are getting `header: false` here, it means that the coValues related to the invitation haven't been uploaded yet to the sync server.

With this new logic we are going to get an "unavailable" here and the CoValueState is going to retry the load.

This doesn't solve our problem with invitations in general because the coValues might not be uploaded for a range of reasons, but should fix the most common issue related to the timing of sharing.

Reply:
The peer responded with both known and content messages bundled in one single response and as I remember the known msg had header: false , whereas content msg was complete. And that situation isn't right as I understand it now because the covalue was actually known and the header had to be provided.
I'll try to reproduce it now

Reply:
Yeah, definitely a bug!

Reply:
Ok, looking again at the code, I should have fixed that bug.

When handling a new content [we flag the coValue as available](https://github.com/gardencmp/jazz/blob/5724735a9539fb25a6fbb5ea386ede1c62a161b1/packages/cojson/src/sync.ts#L493) if it wasn't before.

This may happen after that the same peer flagged the coValue as unavailable.

I've noticed this possible flow and covered it with a test [here](https://github.com/gardencmp/jazz/blob/5724735a9539fb25a6fbb5ea386ede1c62a161b1/packages/cojson/src/tests/coValueState.test.ts#L260).

The new CoValueState is designed to have an explicit "unavailable" state and manage the returned value to be always consistent with the current internal state.

Reply:
Perfect, this is an edge case I was worried about! Thanks for ensuring it's handled correctly.